### PR TITLE
fix(iot-service): Add xml comments to public types and methods

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: iothub/service/src/Common/Extensions/EnumerableExtensions.cs [hunk 1]
```diff
@@ -1,40 +1,32 @@
...
         public static IList<T> ToListSlim<T>(this IEnumerable<T> enumerable, bool requireMutable = false)
         {
-            var list = enumerable as IList<T>;
-            if (list != null && !(requireMutable && list.IsReadOnly))
-            {
-                return list;
-            }
-
-            return enumerable.ToList();
+            return enumerable is IList<T> list && !(requireMutable && list.IsReadOnly)
```
Comment (drwill-ms) [inline iothub/service/src/Common/Extensions/EnumerableExtensions.cs:27]: if enumable is null, will it fail the "is IList<T>" check? If not, we lost our NRE guard.

File: iothub/service/src/Common/Extensions/ExceptionExtensions.cs [hunk 2]
```diff
@@ -121,16 +162,16 @@ public static string ToStringSlim(this Exception exception)
             return exception.ToString();
         }
 
+        /// <summary>
+        /// Gets the exception ID from the given exception.
```
Comment (drwill-ms) [inline iothub/service/src/Common/Extensions/ExceptionExtensions.cs:166]: Prefer casing of Id

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/Common/Extensions/EnumerableExtensions.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/service/src/Common/Extensions/EnumerableExtensions.cs:27]: this being an extension method "enumerable" will never be null, so I think we're good here

File: iothub/service/src/Common/Extensions/ExceptionExtensions.cs (hunk 2 above)
Comment (abhipsaMisra) [reply iothub/service/src/Common/Extensions/ExceptionExtensions.cs:166]: will do

File: iothub/service/src/Common/Extensions/ExceptionExtensions.cs [hunk 3]
```diff
@@ -121,16 +162,16 @@ public static string ToStringSlim(this Exception exception)
...
         {
-            if (exception.Data != null && exception.Data.Contains(ExceptionIdentifierName))
-            {
-                return (string)exception.Data[ExceptionIdentifierName];
-            }
-            else
-            {
-                return null;
-            }
+            return exception.Data != null && exception.Data.Contains(ExceptionIdentifierName)
```
[the remark this answers is not in the document]
Comment (abhipsaMisra) [reply iothub/service/src/Common/Extensions/ExceptionExtensions.cs:172]: this is also an extension method, so the null check for "exception" wouldn't add value

**Pull request merged**